tenzoMath/newRM.h

```diff
@@ -0,0 +1,60 @@
+#ifndef NEW_RM
+#define NEW_RM
+#include <vector>
+#include <opencv2/core.hpp>
+#include <array>
+
+/**
+* \brief class for matematics of robot manipulator based on Denavit-Hartenberg parameters
+*/
+class RoboModel
+{
+public:
+    /**
+    * \brief struct of D-H parameters
+    *
+    * d: offset along previous z to the common normal;
+    * q: angle about previous  z, from old  x to new  x;
+    * a: offset along x in current frame;
+    * alpha: angle about x in current frame;
+    */
+    struct DhParameters
+    {
+        double _dParam;
+        double _qParam;
+        double _aParam;
+        double _alphaParam;
+
+        DhParameters(const double d, const double q, const double a, const double alpha);
```

Comment:
(**must be changed**): I didn't find definition of this constructor. If it's not exist write constructor.